PWGHF/Utils/UtilsBfieldCCDB.h

```diff
@@ -0,0 +1,62 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file UtilsBfieldCCDB.h
+/// \brief Utility to set the B field in analysis querying it from CCDB
+/// \author Mattia Faggin <mfaggin@cern.ch>, University and INFN Padova, Italy
+
+#include "Framework/AnalysisDataModel.h"
+#include "ReconstructionDataFormats/Vertex.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "DataFormatsParameters/GRPObject.h"
+#include "DetectorsBase/Propagator.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "DataFormatsParameters/GRPMagField.h"
+#include "Framework/AnalysisHelpers.h"
+
+/// @brief function to setup the grp object from magnetic field (w/o matCorr for propagation)
+/// @param bc is the bunch crossing
+/// @param mRunNumber is an int with the run umber of the prvious iteration. If at the current iteration it changes, then the grp object is updated
+/// @param ccdb is the o2::ccdb::BasicCCDBManager object
+/// @param ccdbPathGrp is the path where the GRP oject is stored
+/// @param lut is a pointer to the o2::base::MatLayerCylSet object
+/// @param isRun2 tells whether we are analysing Run2 converted data or not (different GRP object type)
+void initCCDB(o2::aod::BCsWithTimestamps::iterator const& bc, int& mRunNumber,
+              o2::framework::Service<o2::ccdb::BasicCCDBManager>& ccdb, std::string ccdbPathGrp, o2::base::MatLayerCylSet* lut,
+              bool isRun2)
+{
+
+  if (mRunNumber != bc.runNumber()) {
+
+    LOG(info) << "====== initCCDB function called (isRun2==" << isRun2 << ")";
+    if (isRun2) { // Run 2 GRP object
+      o2::parameters::GRPObject* grpo = ccdb->getForTimeStamp<o2::parameters::GRPObject>(ccdbPathGrp, bc.timestamp());
+      if (grpo != nullptr) {
```

Comment:
It would be cleaner and simpler if you check `if (grpo == nullptr)` and throw the `fatal`. Then you don't need another block for the rest of the code.